### PR TITLE
USHIFT-4536: Add Gateway API tests

### PR DESCRIPTION
**Pull request by pacevedom**
**Which issue(s) this PR addresses**:

Closes #<Issue Number>

### CODE REVIEW

**Comment by pacevedom** [comment]
/hold
Waiting on https://github.com/openshift/microshift/pull/4066
Will be rebased after that one merges, commits are only included for CI testing.

File: test/suites/standard2/gateway-api.robot [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+Deploy Hello MicroShift
+    [Documentation]    Deploys the hello microshift application (service included)
+    ...    in the given namespace.
+    Create Hello MicroShift Pod    ns=${NS_GATEWAY_1}
+    Expose Hello MicroShift    ${NS_GATEWAY_1}
+
+Setup Namespaces
+    [Documentation]    Configure the required namespaces for namespace ownership tests.
+    Set Suite Variable    \${NS_GATEWAY_1}    ${NAMESPACE}-gw-1
+    Create Namespace    ${NS_GATEWAY_1}
```
Comment (pmtk) [inline test/suites/standard2/gateway-api.robot:41]: Documentation says "configure ... namespaces" but the keyword only creates a new namespace. Do we need the extra namespace?

File: test/suites/standard2/gateway-api.robot [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+
+*** Test Cases ***
+Test Simple HTTP Route
+    [Documentation]    Create a gateway and test it with Hello MicroShift application with HTTPRoute.
+    [Setup]    Run Keywords
+    ...    Setup Namespaces
+    ...    Deploy Hello MicroShift
+    Create Gateway    hostname=${GATEWAY_1_HOSTNAME}    port=${GATEWAY_1_PORT}    namespace=${NS_GATEWAY_1}
+    Create HTTP Route    hostname=${GATEWAY_1_HOSTNAME}    namespace=${NS_GATEWAY_1}
```
Comment (pmtk) [inline test/suites/standard2/gateway-api.robot:25]: We usually don't reference all the args by name, right? Is there a technical reason for this?

File: test/suites/standard2/gateway-api.robot [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    [Documentation]    Create a gateway using given hostname and port. Waits for readiness
+    [Arguments]    ${hostname}    ${port}    ${namespace}
+    ${tmp}=    Set Variable    /tmp/gateway.yaml
+    Set Test Variable    ${HOSTNAME}    ${hostname}
+    Set Test Variable    ${PORT}    ${port}
+    Run Keyword And Ignore Error
+    ...    Remove File    ${tmp}
+    Generate File From Template    ${GATEWAY_MANIFEST_TMPL}    ${tmp}
+    Oc Apply    -n ${namespace} -f ${tmp}
+    Run With Kubeconfig    oc wait -n ${namespace} gateway/test-gateway --for="condition=Accepted" --timeout=60s
```
Comment (pmtk) [inline test/suites/standard2/gateway-api.robot:57]: We do have an `Oc Wait` keyword - would it work?

```suggestion
    Oc Wait    -n ${namespace} gateway/test-gateway --for="condition=Accepted" --timeout=60s
```

**Pull request merged**